src/site/twitch.tv/modules/media-player/MediaPlayer.vue

```diff
@@ -0,0 +1,204 @@
+	</template>
+</template>
+
+<script setup lang="ts">
+import { onUnmounted, ref, toRef, watch } from "vue";
+import { isNumber } from "@vueuse/core";
+import { HookedInstance } from "@/common/ReactHooks";
+import {
+	defineNamedEventHandler,
+	definePropertyHook,
+	unsetNamedEventHandler,
+	unsetPropertyHook,
+} from "@/common/Reflection";
+import { useConfig } from "@/composable/useSettings";
+import { useTooltip } from "@/composable/useTooltip";
+import ForwardIcon from "@/assets/svg/icons/ForwardIcon.vue";
+import GaugeIcon from "@/assets/svg/icons/GaugeIcon.vue";
+import VideoStatsTooltip from "./VideoStatsTooltip.vue";
+
+const props = defineProps<{
+	instance: HookedInstance<Twitch.MediaPlayerComponent>;
+	videoStatsTeleportLocation: HTMLElement | undefined;
+}>();
+
+const teleportLocation = toRef(props, "videoStatsTeleportLocation");
+const doTeleport = ref<boolean>(false);
+
+const shouldShowVideoStats = useConfig<boolean>("player.video_stats");
+const actionOnClick = useConfig<number>("player.action_onclick");
+
+const mediaPlayerProps = ref<Twitch.MediaPlayerComponent["props"]>();
+// const statePath = ref<string>();
+const videoStats = ref({
+	liveLatency: "-.--",
+	droppedFrames: 0,
+	playbackRate: 0,
+	bitrate: "0",
+	width: 0,
+	height: 0,
+	framerate: 0,
+});
+
+if (props.instance.component.props) {
+	definePropertyHook(props.instance.component, "props", {
+		value: (v: Twitch.MediaPlayerComponent["props"]) => {
+			mediaPlayerProps.value = v;
+			if (v.mediaPlayerInstance) {
+				definePropertyHook(v.mediaPlayerInstance.core, "state", {
+					value: (v) => {
+						definePropertyHook(v, "liveLatency", {
+							value: (v) => {
+								if (isNumber(v)) videoStats.value.liveLatency = v?.toFixed(2) ?? "-.--";
+							},
+						});
+						definePropertyHook(v, "statistics", {
+							value: (v) => {
+								definePropertyHook(v, "droppedFrames", {
+									value: (v) => {
+										if (isNumber(v)) videoStats.value.droppedFrames = v;
+									},
+								});
+								definePropertyHook(v, "bitrate", {
+									value: (v) => {
+										if (isNumber(v)) videoStats.value.bitrate = (v / 1000)?.toFixed(0) ?? "0";
+									},
+								});
+								definePropertyHook(v, "framerate", {
+									value: (v) => {
+										if (isNumber(v)) videoStats.value.framerate = v;
+									},
+								});
+							},
+						});
+						definePropertyHook(v, "playbackRate", {
+							value: (v) => {
+								// Twitch automatically applies a 1.03x speed up when latency is large
+								if (isNumber(v)) videoStats.value.playbackRate = v;
+							},
+						});
+						definePropertyHook(v, "quality", {
+							value: (v) => {
+								definePropertyHook(v, "height", {
+									value: (v) => {
+										if (isNumber(v)) videoStats.value.height = v;
+									},
+								});
+								definePropertyHook(v, "width", {
+									value: (v) => {
+										if (isNumber(v)) videoStats.value.width = v;
+									},
+								});
+							},
+						});
```

Comment:
None of these deeper hooks are necessary. Hooking `props` will emit a value update for any nested changes. Simply assign these values to refs, vue will handle the diffing for you.

Reply:
This is not the behavior I'm seeing. The values under the `state` property change far more frequently then the `props`. I have to create those deeper hooks in order to get them to update as often as they change. Is there another way to go about this?